packages/xo-vmdk-to-vhd/src/vmdk-read.js

```diff
@@ -306,25 +305,57 @@ export class VMDKDirectParser {
   }
 }
 
+/**
+ *
+ * the grain table is the array of LBAs (in byte, not in sector) ordered by their position in the VDMK file
+ * THIS CODE RUNS ON THE BROWSER
+ */
 export async function readVmdkGrainTable (fileAccessor) {
-  let headerBuffer = await fileAccessor(0, 512)
-  let grainAddrBuffer = headerBuffer.slice(56, 56 + 8)
+  const getLongLong = (buffer, offset, name) => {
+    if (buffer.length < offset + 8) {
+      throw new Error(
+        `buffer ${name} is too short, expecting ${offset + 8} minimum, got ${
+          buffer.length
+        }`
+      )
+    }
+    const res = new DataView(buffer).getUint32(offset, true)
```

Comment:
I don't understand why you sometimes use `Buffer`'s methods, and sometimes `DataView`'s, is there a reason?

Reply:
I should not have used Buffer's stuff here, it might have slipped, can you point them to me please?